documentation/import-from-rdbms.adoc

```diff
@@ -0,0 +1,120 @@
+====
+
+Once the connection is successful, you'll be prompted to review and confirm the table schema for the database you've chosen to connect to. Here you can see a summary of the tables and columns in your database to help validate you're connected to the right data source. If everything looks good, hit `Confirm`. This will add your data source and and save it for future re-use.
+
+image::import-source-table-schema.png[]
+
+[TIP]
+====
+While retain your connection details, we don't currently retain your password. You'll need to re-provide the password for your data source when you use it again or trigger the import job.
+====
+
+Once you've added a data source, you can either start generating a candidate graph from it, or hit `Later` to come back and do this at a later point in time. If you come back later, you'll be able to use the `...` button on a row and the `New graph model from data source` option to start the graph generation flow. In this guide, let's move immediately onto the next stage of generating a graph model.
+
+
+== Generating a graph model
+
+
+After adding a data source, hit the `Generate graph model` button. Doing this inspects any constraints that exist in your database, looking for primary key constraints to create nodes with IDs and foreign keys to generate relationships between them. If suitable contraints are found we'll generate a corresponding candidate graph for and map the tables and columns to it. If your dataset doesn't have any such constraints (or only part of the schema does), you'll have to map your tables to nodes and relationships manually. If you're not familiar with your data source, it might take some time to understand the different entities and connections present in your tables.
+
+If we don't find any constraints *at all* in your source database, we'll map everyhthing to a node for your convenience. While this saves you doing some mapping manually, please note that a graph with just nodes doesn't make for a connected graph! From here, delete nodes that should be relationships and recreate relationships between the remaining nodes as is most appropriate for your data.
+
+
+[TIP]
+=====
+The generation and mapping doesn't currently support composite primary or foregin keys, so you won't see mappings related to these, nor be able to import them easily via the import UI just yet.
+=====
+
+=== Refining a model
+
+Even in this example where a well-conencted graph model has been generated, the model can benefit from some tidy up. For example, tables mapped to nodes simply use the correspoinding table name as the Label, but we can tidy that up to make it more suitable for our Graph. The same applies for relatonships which by default use a concatenation of related table names. Relationship direction may also benefit from being reversed for the chosen semantics. See the example below where the concepts of Person, Order and Employee and their corresponding relationships are tidied up.
```

Comment:
```suggestion
Even in this example where a well-connected graph model has been generated, the model can benefit from some tidy-up. 
For example, tables mapped to nodes use the corresponding table name as the label, but this can be made more suitable for your graph. 
The same applies to relationships that by default use a concatenation of related table names. 
Relationship direction may also benefit from being reversed for the chosen semantics. 
See the example below where the concepts of Person, Order, and Employee and their corresponding relationships are tidied up.
```